api.proto

```diff
@@ -214,21 +214,33 @@ message StreamMetadata {
         OK             = 0;
         UNKNOWN_STREAM = 1;
     }
-    string                        name       = 1; // The name of the stream being described
-    string                        subject    = 2; // The stream subject
-    Error                         error      = 3; // Indicates if there was something wrong with the requested stream
-    map<int32, PartitionMetadata> partitions = 4; // Information for the stream partitions
+    string                        name              = 1; // The name of the stream being described
+    string                        subject           = 2; // The stream subject
+    Error                         error             = 3; // Indicates if there was something wrong with the requested stream
+    map<int32, PartitionMetadata> partitions        = 4; // Information for the stream partitions
+    int64                         creationTimestamp = 5 [jstype=JS_STRING]; // When the stream has been created
+}
+
+// PartitionEventMetrics contains metrics related to a stream partition.
+message PartitionEventMetrics {
+    int64 firstTimestamp = 1 [jstype=JS_STRING]; // When the first event has occured
+    int64 lastTimestamp  = 2 [jstype=JS_STRING]; // When the last event has occured
 }
 
 // PartitionMetadata contains information for a stream partition.
 message PartitionMetadata {
-    int32            id            = 1; // Partition id
-    string           leader        = 2; // Broker id of the partition leader
-    repeated string  replicas      = 3; // Broker ids of the partition replicas
-    repeated string  isr           = 4; // Broker ids of the in-sync replica set
-    int64            highWatermark = 5; // High watermark of the partition
-    int64            newestOffset  = 6; // Newest offset of the partition
-    bool             paused        = 7; // Is this partition paused?
+    int32                 id                = 1; // Partition id
+    string                leader            = 2; // Broker id of the partition leader
+    repeated string       replicas          = 3; // Broker ids of the partition replicas
+    repeated string       isr               = 4; // Broker ids of the in-sync replica set
+    int64                 highWatermark     = 5; // High watermark of the partition
+    int64                 newestOffset      = 6; // Newest offset of the partition
+    bool                  paused            = 7; // Is this partition paused?
+    bool                  readonly          = 8; // Is this partition read-only?
+    int64                 creationTimestamp = 9 [jstype=JS_STRING]; // When the partition has been created
+    PartitionEventMetrics messageMetrics    = 10; // Metrics for received messages
+    PartitionEventMetrics pauseMetrics      = 11; // Metrics for paused status
+    PartitionEventMetrics readonlyMetrics   = 12; // Metrics for readonly status
```

Comment:
I'm confused what PartitionEventMetrics mean for pause and readonly. Are they when the event was processed?

Reply:
Yes, their are only used to store the first and last time the pause and readonly status has been changed. They could also contain the number of times this change has been done, if we believe that could be a useful information.

Reply:
might be worth a comment giving that context

Reply:
Fixed in 143245f.